src/main/Tree.razor.cs

```diff
@@ -190,54 +190,47 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
                 }
                 bool urlSet = false;
                 var query = QueryHelpers.ParseQuery(uri.Query);
-                if (query.TryGetValue("avatarUrl", out var encodedAvatarUrl))
+                if (TreeQuery.TryParse(uri.Query, out var encodedAvatarUrl))
                 {
-                    Uri uriResult;
-                    string decodedUrl = Nancy.Helpers.HttpUtility.UrlDecode(encodedAvatarUrl);
-                    bool validUrl = Uri.TryCreate(decodedUrl, UriKind.Absolute, out uriResult)
-                        && (uriResult.Scheme == Uri.UriSchemeHttp || uriResult.Scheme == Uri.UriSchemeHttps);
-                    if (validUrl)
+                    if (encodedAvatarUrl != null)
```

Comment:
This if condition is unnecessary since TryParse will only return true if encodedAvatarUrl is not null.